### PR TITLE
Feature: Added 254 Civilian Professions

### PR DESCRIPTION
**Pull request by IllianiBird**
This PR adds 254 Civilian professions. Each profession has 2 tied skills, meaning that it's possible to improve the experience level of your Janitor! Jokes aside, there are some pretty useful skill combinations and this is being used as a way to make dependents more useful.

Now, whenever a dependent is generated they'll generate with a profession (or as unemployed, that's possible too). This does mean that you're expected to pay them. However, the assumption is that some will be kept around for roleplay purposes (your chefs, janitors, personal valets, clowns) but most will be immediately put to work on more useful tasks.

I'll include a full list of professions below but will make one note: the Adult Entertainer and Luxury Companion roles have a very strict age limit and we've put extensive testing in place to ensure it cannot be circumvented.

- Unemployed (was 'Dependent')
- Unspecified Profession (for those Dependents you don't want flagged as 'Unemployed' just requires Career/Any)
- Adult Entertainer
- AeroSpace Scavenger
- Agri-Farmer
- Aide-de-camp
- Antiquarian
- Arms Dealer
- Artisan Microbrewer
- Artist
- Astech Trainer
- Astrographer
- Banquet Planner
- Barber
- Bartender
- BattleROM Editor
- Bodyguard
- Brawler
- Broker
- Burglar
- Cargo Seal Inspector
- Chef
- Civic Controller
- Civil Clerk
- Civilian AeroSpace Instructor
- Civilian AeroTek
- Civilian Debriefer
- Civilian Doctor
- Civilian DropShip Pilot
- Civilian Guard
- Civilian JumpShip Navigator
- Civilian Liaison
- Civilian Pilot
- Civilian Requisition Officer
- Civilian VTOL Pilot
- Claims Arbitrator
- Clown
- Codebreaker
- Combat Chaplain
- Combat Prosthetics Fitter
- Comms Operator
- Community Leader
- Con Artist
- Conflict Resolution Trainer
- Corporate Concierge
- Corporate Espionage Agent
- Counterfeiter
- Counterintelligence Liaison
- Courier
- Criminal Mechanic
- Cultist
- Cultural Censor
- Cultural Liaison
- Cultural Sensitivity Advisor
- Customs Inspector
- Data Analyst
- Data Harvester
- Data Intercept Operator
- Data Launderer
- Data Leak Tracker
- Data Smuggler
- Decommissioning Specialist
- Deep Cover Operator
- Disease Control Administrator
- Disgraced Noble
- Doomsday Preacher
- DropShip Entertainment Officer
- Drug Dealer
- Duelist
- Early Detection Systems Operator
- Election Fixer
- Embedded Military Holo-Filmer
- Escaped Prisoner
- Exoskeleton Laborer
- Exotic Pet Caretaker
- Factory Foreman
- Factory Worker
- False Prophet
- Fence
- Field Hospital Administrator
- Field Laundry Operator
- Firefighter
- Firing Range Safety Officer
- Fisher
- Food Truck Operator
- Freight Lift Operator
- Frontier Doctor
- Gambler
- Gang Leader
- Genetic Therapy Specialist
- Geneticist
- Graffiti Artist
- Guild Liaison
- Hacker
- Historian
- Holo-Cartographer
- Holo-DJ
- Holo-Gamer
- Holo-Journalist
- Holo-Star
- Identity Fabricator
- Illegal Pet Smuggler
- Implant Surgeon
- Industrial Inspector
- Industrial Mek Pilot
- Information Broker
- Intelligence Analyst
- Interior Decorator
- Interrogator
- Interstellar Tourism Guide
- Janitor
- JumpShip Botanist
- JumpShip Chef
- Lawyer
- Legal Archivist
- Librarian
- Livestock Farmer
- Livestream Entertainer
- Local Warlord
- Logistics Coordinator
- Lorekeeper
- Loyalty Auditor
- Loyalty Monitor
- Luxury Companion
- Market Manipulator
- Masseuse
- Media Manipulator
- Mediator
- Medical Researcher
- Mek Range Instructor
- MekBay Scheduler
- Merchant
- Mess Hall Manager
- Military Academy Dropout
- Military Accountant
- Military Analyst
- Military Contractor
- Military Coroner
- Military Liaison
- Military Painter
- Military Pony Express Courier
- Military Promoter
- Military Recruiter
- Military Tattoo Artist
- Military Theorist
- Militia Leader
- Minefield
- Miner
- Morale Officer
- Mountain Climber
- Munitions Clerk
- Munitions Factor Worker
- Musician
- Mythologist
- Neurohelmet Interface Calibrator
- Noble Heir in Hiding
- Noble Page
- Orbital Debris Tracker
- Orbital Defense Gunner
- Orbital Shuttle Pilot
- Organ Harvester
- Painter
- Paramedic
- Paramilitary Trainer
- Pathfinder
- Penniless Noble
- Performer
- Personal Archivist
- Personal Assistant
- Personal Herald
- Personal Sommelier
- Personal Valet
- Philosopher
- Physical Rehabilitation Therapist
- Planetary Adaptation Physiologist
- Planetary Immigration Processor
- Planetary Migration Coordinator
- Planetary Surveyor
- Political Agitator
- Prison Fixer
- Prison Guard
- Professional Cosplayer
- Propagandist
- Prophet
- Protection Racketeer
- Psychologist
- PsyOps Broadcaster
- Public Execution Broadcaster
- Public Transport Overseer
- Radiation Risk Monitor
- Ration Distributor
- Recruitment Screen Officer
- Reeducation Specialist
- Relationship Matchmaker
- Relic Hunter
- Religious Leader
- Repair Bay Supervisor
- Revolutionist
- Riot Response Planner
- Ritualist
- Salvage Rat
- Scandal Fixer
- Scribe
- Sculpturer
- Security Advisor
- Security Desk Operator
- Sensor Technician
- Signal Jamming Specialist
- Smuggler
- Spaceport Bureaucrat
- Spaceport Courier
- Spaceport Engineer
- Spaceport Worker
- Sports Star
- Spy
- Steward
- Street Surgeon
- Subversive Poet
- Surgical Simulator Instructor
- Surveillance Expert
- Surveillance Sweeper
- Swimming Instructor
- Systems Consultant
- Tactical Analyst
- Tailor
- Tax Auditor
- Teacher
- Tech/Air Filtration
- Tech/Communications
- Tech/Encryption
- Tech/Fusion Plant
- Tech/Hydroponics
- Tech/Security
- Tech/Waste Management
- Tech/Water Reclamation
- Tech/Zero-G
- Thief
- Thug
- Torturer
- Training Sim Designer
- Training Sim Operator
- Transport Driver
- Trauma Counselor
- Unlicensed Chemist
- Virtual Reality Therapist
- VR Entertainer
- War Correspondent
- War Crime Investigator
- Warehouse Worker
- Warfare Planner
- Weapons Tester
- Weathercaster
- Wedding Planner
- Xenoanimal Trainer
- Xenobiologist
- Zero-G Physical Therapist

### CODE REVIEW

**Comment by HammerGS** [comment]
Medical Specialties

Neural Implant Calibration Technician - Skills: S_COMPUTERS, S_SCIENCE_BIOLOGY
Fine-tunes neural interfaces and medical implants for optimal function and minimal side effects

Genetic Screening Counselor - Skills: S_SCIENCE_GENETICS, S_SCIENCE_PSYCHOLOGY
Interprets genetic testing results and advises individuals on potential hereditary conditions

Artificial Organ Maintenance Specialist - Skills: S_TECH_MECHANIC, S_SCIENCE_BIOLOGY
Services and adjusts cybernetic and vat-grown replacement organs

Planetary Adaptation Physiologist - Skills: S_SCIENCE_BIOLOGY, S_SCIENCE_PSYCHOLOGY
Helps people adapt to different planetary conditions (gravity, atmosphere, day length)

Medical Data Analyst - Skills: S_COMPUTERS, S_SCIENCE_MATHEMATICS
Processes patient outcomes and epidemic patterns across populations

Trauma Memory Restructuring Therapist - Skills: S_SCIENCE_PSYCHOLOGY, S_COMPUTERS
Uses neural interface technology to help patients process and integrate traumatic memories

Battlefield Prosthetic Designer - Skills: S_TECH_MECHANIC, S_SCIENCE_BIOLOGY
Creates specialized limb replacements optimized for various planetary conditions

Zero-G Physical Adaptation Specialist - Skills: S_ZERO_G_OPERATIONS, S_SCIENCE_BIOLOGY
Helps space travelers and residents prevent and recover from microgravity effects

Pharmaceutical Formulation Technician - Skills: S_SCIENCE_PHARMACOLOGY, S_SCIENCE_CHEMISTRY
Customizes medications for individuals with unique genetic profiles or for planetary conditions

Biofeedback Equipment Technician - Skills: S_TECH_MECHANIC, S_SCIENCE_PSYCHOLOGY
Maintains and calibrates equipment used in psychological treatment and stress management

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 1]
```diff
@@ -3663,217 +3714,113 @@
~    }
~
~    public SkillLevel getSkillLevel(final Campaign campaign, final boolean secondary) {
         return Skills.SKILL_LEVELS[getExperienceLevel(campaign, secondary) + 1];
     }
 
+    /**
```
[the remark this answers is not in the document]
Comment (IllianiBird) [reply MekHQ/src/mekhq/campaign/personnel/Person.java:3720]: Prep for the upcoming integration of Attribute score modifiers

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 2]
```diff
@@ -3663,217 +3714,113 @@
...
+     * Determines the experience level of a person in their current profession within the context of a campaign.
+     *
```
[the remark this answers is not in the document]
Comment (IllianiBird) [reply MekHQ/src/mekhq/campaign/personnel/Person.java:3722]: Prep for the upcoming integration of Attribute score modifiers

File: MekHQ/src/mekhq/campaign/personnel/Person.java [hunk 3]
```diff
@@ -3882,14 +3829,112 @@
...
+        Skill skill = getSkill(skillNames.get(0));
+        if (skill == null) {
+            return SkillType.EXP_NONE;
+        }
+
+        return skill.getType().getExperienceLevel(averageSkillLevel);
+    }
+
+
+    private int CalculateExperienceLevelForProfession(String sGunBa, String sAntiMek,
```
[the remark this answers is not in the document]
Comment (IllianiBird) [reply MekHQ/src/mekhq/campaign/personnel/Person.java:3924]: Good bot
